rest/src/main/java/org/jboss/pnc/rest/provider/BuildRecordProvider.java

```diff
@@ -740,11 +789,18 @@ public BuildRecordRest getLatestBuildRecord(Integer configId, boolean executedOn
         PageInfo pageInfo = new DefaultPageInfo(pageIndex * pageSize - offset, pageSize);
         SortInfo sortInfo = sortInfoProducer.getSortInfo(sort);
 
-
-        List<BuildRecordRest> content = nullableStreamOf(((BuildRecordRepository) repository)
+        List<BuildRecord> buildRecords = nullableStreamOf(((BuildRecordRepository) repository)
                 .queryWithPredicatesUsingCursor(pageInfo, sortInfo, dbAndPredicates, dbOrPredicates))
-                .map(toRESTModel())
                 .collect(Collectors.toList());
+        Set<IdRev> buildRecordsIdRevs = nullableStreamOf(buildRecords).map(br -> new IdRev(br.getBuildConfigurationId(), br.getBuildConfigurationRev())).collect(Collectors.toSet());
+
+        List<BuildRecordRest> content = new ArrayList<>();
+        if (!buildRecordsIdRevs.isEmpty()) {
+            Map<IdRev, BuildConfigurationAudited> buildConfigurationsAuditedMap = buildConfigurationAuditedRepository.queryById(buildRecordsIdRevs);
+            content = nullableStreamOf(buildRecords).map(toRESTModel(buildConfigurationsAuditedMap))
```

Comment:
Old method toRESTModel() invoked **buildConfigurationAuditedRepository.queryById(IdRev)**, so same comments as above are valid, and similar code changes applied.